src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommand.cs

```diff
@@ -43,16 +43,26 @@ internal static class ListPackageCommand
                     Strings.ListPkg_FrameworkDescription,
                     CommandOptionType.MultipleValue);
 
-                var includeOutdated = listpkg.Option(
+                var outdatedReport = listpkg.Option(
                     "--outdated",
                     Strings.ListPkg_OutdatedDescription,
                     CommandOptionType.NoValue);
 
-                var includeDeprecated = listpkg.Option(
+                var deprecatedReport = listpkg.Option(
                     "--deprecated",
                     Strings.ListPkg_DeprecatedDescription,
                     CommandOptionType.NoValue);
 
+                var vulnerableReport = listpkg.Option(
+                    "--vulnerable",
+                    Strings.ListPkg_VulnerableDescription,
+                    CommandOptionType.NoValue);
+
+                var isOffline = listpkg.Option(
```

Comment:
> Added --offline option to make standard list (i.e. not --vulnerable, --deprecated, or --outdated reports) exhibit pre-existing behavior of not accessing server metadata

Honestly `--offline` is pretty damn weird. 
Not sure how thoroughly we reviewed the spec last time, but isn't this a breaking experience? 
`dotnet list package` remaining offline makes sense to me.

Reply:
I think `--offline` was my suggestion, so I'm obviously biased, but I think it's a great idea. In VS, PM UI checks for outdated and deprecated packages without the user having to explicitly request it. Why shouldn't the command line version?

We don't have telemetry for CLI tools, so we can't make a data based decision, at best we can do customer interviews. But if we think that most customers would find outdated, deprecated and vulnerability information useful, then `dotnet list package` should check it by default.

Is it a breaking change? Yes, there are definitely scenarios that customers could be using `dotnet list package` today that will stop working if we make ANY change to the output. For offline scenarios, checking sources will break with the error that Drew reported further up, which does not happen with .NET Core SDK 3.1 and earlier.  But if this can be done quickly enough to make it into .NET 5, a major version increment, and we believe it has a better customer experience, is it worthwhile?

If we're worried about people who call `dotnet list package` in a script and parse the results, and this breaking them, then we need to prioritise the work to implement the "machine readable" (json) output.

Reply:
IMO, the list command shouldn’t error out if there is no internet connectivity.

Reply:
/cc: @JonDouglas

Reply:
It seems that these are concerns about the spec itself, not this PR. Clearly this needs to be resolved. I'll be reaching out.

Reply:
I don't believe we need an `--offline` command. But then again, it would need to be replaced with something. Either an offline cache or a quick network check to determine the code path. If I can `dotnet restore` against a local package source, I should be able to `dotnet list package` against a local environment with or without internet connection & no additional args.

Reply:
Fwiw restore is different in that it assumes package immutability. 
That means if a package is *installed* in the global packages folder, we won't bother. 

Things such as deprecation and vulnerability have an explicit point in time dependency, so it only uses the http cache information. 
The http cache has a 30 min expiry time, and there are no fallbacks.

See related info about restore + sources: https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#packagereference-and-sources

Reply:
That's kinda where I'm going with it all. If we attach an online functionality to `dotnet list package`, it needs to then have an offline capability.

What realistically makes sense to me is having a `dotnet nuget audit` command which does all this: it gives you an audited list based on your dependencies & gives you an action to take. You then have the action of `dotnet nuget audit fix` or instructions to manually resolve your issues if we can't do it for you.

Right now that is done via `dotnet list package --<blah>` which is three different things that need to be audited. Because we attach this to the `dotnet list package`, we now have to support offline functionality rather than a new command avoiding this. So if in stage 2 we add a `dotnet nuget audit` command, we're basically doing the same thing at no benefit to the user other than adding noise to the `dotnet list package` command. 

Ideal scenario:

`dotnet list package` - Does what it has done previously & stays offline.
`dotnet nuget audit` - Does what `dotnet list package --vulnerable/outdated/deprecated` does today & provides a resolution follow-up or instructions on how to resolve if it cannot be auto-fixed.
`dotnet restore` - Does a `dotnet nuget audit` implicitly.
`dotnet nuget audit fix` - Attempts to fix results of `dotnet nuget audit`.

However since this work is already in-flight, we're adding various behavior to `dotnet list package` to additionally support the same functionality `dotnet nuget audit` will have & on detection of a vulnerable/deprecated/outdated package, should provide the user a logical next step to resolve via `dotnet nuget audit fix` or a manual review of potential solutions.

In the case of a user having all three audits in a `dotnet list package`, they have to type three commands just to get a further idea of resolution. This could be saving keystrokes by only having one `audit` command instead.

Reply:
> dotnet list package remaining offline makes sense to me.

@nkolev92 , since --outdated and --deprecated already already require network connection, we can no longer say dotnet list package shouldn't use network... right? Correct me if I'm wrong. 

> IMO, the list command shouldn’t error out if there is no internet connectivity.

@anangaur , agree. Perhaps we can fail gracefully and show a message that deprecation and vulnerability information are unavailable without network.

@JonDouglas , I agree that dotnet list audit will be the cleanest experience, and we should add it as part of stage 2. Regarding stage 1, do you have any concerns regarding the spec as it's documented here: https://github.com/NuGet/Home/wiki/dotnet-list-package---vulnerable 
if so, let's schedule a meeting and modify the spec.